llvm/lib/Transforms/Utils/LoopUnroll.cpp

```diff
@@ -1256,14 +1257,19 @@ llvm::canParallelizeReductionWhenUnrolling(PHINode &Phi, Loop *L,
     return std::nullopt;
   RecurKind RK = RdxDesc.getRecurrenceKind();
   // Skip unsupported reductions.
-  // TODO: Handle additional reductions, including FP and min-max
-  // reductions.
-  if (!RecurrenceDescriptor::isIntegerRecurrenceKind(RK) ||
+  // TODO: Handle additional reductions, including min-max reductions.
+  if (!(RecurrenceDescriptor::isIntegerRecurrenceKind(RK) ||
+        RecurrenceDescriptor::isFloatingPointRecurrenceKind(RK)) ||
```

Comment:
```suggestion
  if (
```

I think the condition is always true, given how `isFloatingPointRecurrenceKind` is implemented, as long as RK != None, which `RecurrenceDescriptor::isReductionPHI` should already ensure

```
bool RecurrenceDescriptor::isFloatingPointRecurrenceKind(RecurKind Kind) {                                <<<
  return (Kind != RecurKind::None) && !isIntegerRecurrenceKind(Kind);
}
```

Reply:
Makes sense, thanks -- done!